File: pset2/caesar.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <cs50.h>
#include <stdlib.h>

int main(int argc, string argv[]){
    if (argc!=2) {
       printf ("Usage: ./caesar k\n");
       return 1;
   }
   else{
       int key = atoi(argv[1]);
   printf("plaintext: ");
   string text = GetString();
   printf("ciphertext: ");

      int len = strlen(text);
       for (int i = 0; i <= len-1; i++){
           if (text[i]>=97 && text[i]<=122){
               char crypt = ((text[i]+key)-97)%26+97;
               printf("%c", crypt);
           }
           else if(text[i]>=65 && text[i]<=90){
               char crypt = ((text[i]+key)-65)%26+65;
               printf("%c", crypt);
           }
           else {
               printf("%c", text[i]);
           }
       }
       printf("\n");
       return 0;

   }
   

}

File: pset6/caesar.py
import cs50
import sys

# check for argument's validation
if len(sys.argv)!=2:
    print("Usage ./saesar key")
    exit(1)
# key must be > 0
elif (int(sys.argv[1])<0):
    print("Key must be possitive")
else:
    # crypt the entire text
    print("plaintext: ", end = '')
    text = cs50.get_string()
    crypt = []
    for i in range(len(text)):
        if (text[i].isupper()):
            crypt.append( chr((((ord(text[i]) - ord('A')) + int(sys.argv[1])) % 26) + ord('A')) )
        else:
            crypt.append( chr((((ord(text[i]) - ord('a')) + int(sys.argv[1])) % 26) + ord('a')) )
    # output encrypted text
    print("ciphertext: ", end = '')
    print("".join(crypt))